common/src/Microsoft.Azure.IIoT.Hub.Module.Client/src/Default/IoTSdkFactory.cs

```diff
@@ -205,471 +225,33 @@ public sealed class IoTSdkFactory : IClientFactory, IDisposable {
             if (string.IsNullOrEmpty(ModuleId)) {
-                if (_cs == null) {
-                    throw new InvalidConfigurationException(
-                        "No connection string for device client specified.");
-                }
-                return DeviceClientAdapter.CreateAsync(product, _cs, DeviceId,
-                    transportSetting, _timeout, RetryPolicy, onError, _logger);
-            }
-            return ModuleClientAdapter.CreateAsync(product, _cs, DeviceId, ModuleId,
-                transportSetting, _timeout, RetryPolicy, onError, _logger);
-        }
-
-        /// <summary>
-        /// Adapts module client to interface
-        /// </summary>
-        public sealed class ModuleClientAdapter : IClient {
-
-            /// <summary>
-            /// Whether the client is closed
-            /// </summary>
-            public bool IsClosed { get; internal set; }
-
-            /// <summary>
-            /// Create client
-            /// </summary>
-            /// <param name="client"></param>
-            internal ModuleClientAdapter(ModuleClient client) {
-                _client = client ??
-                    throw new ArgumentNullException(nameof(client));
-            }
-
-            /// <summary>
-            /// Factory
-            /// </summary>
-            /// <param name="product"></param>
-            /// <param name="cs"></param>
-            /// <param name="deviceId"></param>
-            /// <param name="moduleId"></param>
-            /// <param name="transportSetting"></param>
-            /// <param name="timeout"></param>
-            /// <param name="retry"></param>
-            /// <param name="onConnectionLost"></param>
-            /// <param name="logger"></param>
-            /// <returns></returns>
-            public static async Task<IClient> CreateAsync(string product,
-                IotHubConnectionStringBuilder cs, string deviceId, string moduleId,
-                ITransportSettings transportSetting,
-                TimeSpan timeout, IRetryPolicy retry, Action onConnectionLost,
-                ILogger logger) {
-
-                if (cs == null) {
-                    logger.Information("Running in iotedge context.");
+                if (_mqttClientCs != null) {
+                    return MqttClientAdapter.CreateAsync(
+                        product,
+                        _mqttClientCs,
+                        DeviceId,
+                        _telemetryTopicTemplate,
+                        timeout: TimeSpan.FromSeconds(15),
+                        RetryPolicy,
+                        onError,
+                        _logger);
+                }
+                else if (_deviceClientCs != null) {
+                    return DeviceClientAdapter.CreateAsync(product, _deviceClientCs, DeviceId,
+                        transportSetting, timeout: TimeSpan.FromMinutes(5), RetryPolicy, onError, _logger);
```

Comment:
timeout here is 5 minutes, in the mqtt was 15 seconds. can we handle this systematically? ideally not using magic numbers?

Reply:
Done, what is missing setting this through configuration.